bundles/org.openhab.binding.velux/src/main/java/org/openhab/binding/velux/internal/handler/VeluxBridgeHandler.java

```diff
@@ -957,4 +968,82 @@ public ProductBridgeIndex getProductBridgeIndex(String thingName) {
         }
         return ProductBridgeIndex.UNKNOWN;
     }
+
+    /**
+     * Iterate over all roller shutter type things in the hub, and initialise the dynamic vane position channel if the
+     * respective device supports it.
+     */
+    private void initializeVanePositionChannels() {
+        getThing().getThings().stream()
+                .filter(thing -> VeluxBindingConstants.THING_TYPE_VELUX_ROLLERSHUTTER.equals(thing.getThingTypeUID()))
+                .forEach(thing -> initializeVanePositionChannel(thing));
+    }
+
+    /**
+     * For the given roller shutter type thing, initialise the dynamic vane position channel if the respective device
+     * supports it.
+     */
+    private void initializeVanePositionChannel(Thing thing) {
+        final ChannelUID vaneChannelUID = new ChannelUID(thing.getUID(), VeluxBindingConstants.CHANNEL_VANE_POSITION);
+
+        final VeluxProduct product = existingProducts()
+                .get((new Thing2VeluxActuator(this, vaneChannelUID)).getProductBridgeIndex());
+        if (product.equals(VeluxProduct.UNKNOWN)) {
+            // product unknown
+            return;
+        }
+
+        // predicate to filter the vane position channel
+        final Predicate<Channel> vaneChannelFilter = c -> VeluxBindingConstants.CHANNEL_TYPE_VANE
+                .equals(c.getChannelTypeUID());
+
+        // note: this is an immutable list
+        List<Channel> channels = thing.getChannels();
+
+        // current and required state of the vane channel
+        final boolean vaneChannelExisting = channels.stream().anyMatch(vaneChannelFilter);
+        final boolean vaneChannelRequired = product.supportsVanePosition();
+
+        if (vaneChannelExisting == vaneChannelRequired) {
+            // no change needed
+            return;
+        }
+
+        ThingHandler handler = thing.getHandler();
+        if (!(handler instanceof VeluxHandler)) {
+            // handler is of the wrong wrong type
+            return;
+        }
+        VeluxHandler veluxHandler = (VeluxHandler) handler;
+
+        // make a mutable copy of the original immutable channel list
+        channels = new ArrayList<>(channels);
```

Comment:
Why not moving most of this code in a new method in the thing handler? 
This would avoid overriding methods visibility.

Reply:
Makes sense. Will do

Reply:
Done.